src/server/session.ts

```diff
@@ -2040,4 +2043,50 @@ namespace ts.server {
         response?: {};
         responseRequired?: boolean;
     }
+
+    /* @internal */ // exported for tests only
+    /** Since we will perform the rename in a changed document, line and offset information must be adjusted if the new text adds/deletes newlines. */
+    export function fixupLineAndOffset(oldPosition: protocol.Location, oldText: string, location: number, locationFileName: string, edits: ReadonlyArray<FileTextChanges>): protocol.Location {
+        let { line, offset } = oldPosition;
+        const oldLineStartOfLocation = location - (offset - 1); // offset is 1-based, location is 0-based
+
+        for (const { fileName, textChanges } of edits) {
+            if (fileName !== locationFileName) {
+                continue;
+            }
+
+            // Text change spans are in the *old* document. They apply all at once rather than sequentially.

```

Comment:
Isn't it only the edits *before* the (old) position that apply?

Reply:
Thanks, fixed